spec/solver_spec.rb

```diff
@@ -0,0 +1,58 @@
+require_relative '../solver'
+
+describe Solver do
+  solver = Solver.new
+  describe '#factorial' do
```

Comment:
- [ ] Kindly add a test for negative values where the factorial method will raise an exception as an expectation. This is a project requirement.
``hint``  `expect { resolver.factorial(-5) }.to raise_exception`

Reply:
Thank you for the review.